sql/src/test/java/io/crate/integrationtests/OptimizeTableIntegrationTest.java

```diff
@@ -101,6 +101,24 @@ public void testOptimizeWithParams() throws Exception {
     }
 
     @Test
+    public void testOptimizeWithUpgradeSegments() throws Exception {
+        execute("create table test (id int primary key, name string)");
+        ensureYellow();
+        execute("insert into test (id, name) values (0, 'Trillian'), (1, 'Ford'), (2, 'Zaphod')");
+        execute("select count(*) from test");
+        assertThat((long) response.rows()[0][0], lessThanOrEqualTo(3L));
+
+        execute("delete from test where id=1");
+        refresh();
+        execute("optimize table test with (upgrade_segments=true)");
```

Comment:
I'm not sure if this test makes sense at all. the `testUpgradeSegments` test is already doing a integration test, either move is to here, or just leave and delete this test.